fillAssumed.py

```diff
@@ -69,6 +95,33 @@ def __init__(self,
 
         self.itemLists = [self.prog_items, self.extra_items]
 
+    def get_forced_locations(self):
+        # This is highly ascent specific
+        # Get locations for the key items in each zone
+        # "With the bare minimum items to access the zone, the key items will reachable
+        forced_locations = []
+        loadout = Loadout(self.game)
+
+        # You can't leave zone one without these so no reason to force them
+        for item in forced_zone_items['zone-1']:
+            loadout.append(item)
+
+        for item in forced_zone_items['zone-2']: # TODO shuffle these?
+            # find a location the item could be, and then add it to the loadout
+            locations = self._get_available_locations(loadout)
+            locations = [l for l in locations if l['zone'] == 'zone-2']
+            forced_locations.append([item, locations])
+            loadout.append(item)
+
+        # Repeat with zone 3
+        for item in forced_zone_items['zone-3']: # TODO shuffle these?
+            # find a location the item could be, and then add it to the loadout
+            locations = self._get_available_locations(loadout)
+            locations = [l for l in locations if l['zone'] == 'zone-3']
+            forced_locations.append([item, locations])
+            loadout.append(item)
+        return forced_locations
+
```

Comment:
This function creates possible locations for the "key" items in each zone. Zone-1 is handled by the normal seed generator. The result is a with the item and possible locations for this item. My thought on the `ascent_fix = 'duplicate'` option is that you'd call this, check and see if each item is in it's correct zone, and if not, duplicate in a location given by this function.